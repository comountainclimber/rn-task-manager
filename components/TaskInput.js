import React from "react";
import { StyleSheet, View, TextInput, TouchableOpacity } from "react-native";
import { Ionicons } from "@expo/vector-icons";
import PropTypes from "prop-types";

import Card from "./Card";

export default function TaskInput({ handleAddTask }) {
  const [text, onChangeText] = React.useState("");

  return (
    <Card additionalStyles={styles.taskInputContainer}>
      <>
        <TextInput
          placeholder="Add a task"
          onChangeText={value => onChangeText(value)}
          value={text}
          multiline
          style={styles.taskInput}
        />
        <View style={styles.addTaskIconContainer}>
          <TouchableOpacity
            onPress={() => {
              handleAddTask({ value: text });
              onChangeText("");
            }}
          >
            <Ionicons name="ios-add" size={25} color="#B4B4B4" />
          </TouchableOpacity>
        </View>
      </>
    </Card>
  );
}

TaskInput.propTypes = {
  handleAddTask: PropTypes.func.isRequired
};

const styles = StyleSheet.create({
  taskInputContainer: {
    padding: 12
  },
  taskInput: {
    fontFamily: "Nunito-Regular"
  },
  addTaskIconContainer: {
    width: "100%",
    display: "flex",
    alignItems: "flex-end",
    marginBottom: -10
  }
});
