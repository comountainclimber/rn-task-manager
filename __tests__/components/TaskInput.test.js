import React from "react";
import renderer from "react-test-renderer";

import TaskInput from "../../components/TaskInput";

const mock = () => undefined;

it("it renders without crashing", () => {
  const tree = renderer.create(<TaskInput handleAddTask={mock} />).toJSON();
  expect(tree).toMatchSnapshot();
});
