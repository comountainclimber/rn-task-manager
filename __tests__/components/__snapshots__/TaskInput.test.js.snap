// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders without crashing 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#F0F0F0",
        "borderRadius": 4,
        "borderWidth": 1.5,
        "marginBottom": 12,
        "padding": 12,
        "width": "100%",
      },
      Object {
        "padding": 12,
      },
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    blurOnSubmit={true}
    multiline={true}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Add a task"
    rejectResponderTermination={true}
    style={
      Object {
        "fontFamily": "Nunito-Regular",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;
